### PR TITLE
Update frontend poller metrics to include tasklist tag

### PR DESCRIPTION
**Pull request by Shaddoll**
**What changed?**
Update frontend poller metrics to include tasklist tag

**Why?**
Improve observability

**How did you test it?**

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

**Review by timl3136 — APPROVED** [review]
LGTM

**Review by taylanisikdemir — APPROVED** [review]

File: service/frontend/wrappers/metered/api_generated.go [hunk 1]
```diff
@@ -319,9 +319,9 @@ func (h *apiHandler) PollForDecisionTask(ctx context.Context, pp1 *types.PollFor
 	defer func() { log.CapturePanic(recover(), h.logger, &err) }()
 	tags := []tag.Tag{tag.WorkflowHandlerName("PollForDecisionTask")}
 	tags = append(tags, toPollForDecisionTaskRequestTags(pp1)...)
-	scope := h.metricsClient.Scope(metrics.FrontendPollForDecisionTaskScope).Tagged(metrics.DomainTag(pp1.GetDomain())).Tagged(metrics.GetContextTags(ctx)...)
-	scope.IncCounter(metrics.CadenceRequests)
-	sw := scope.StartTimer(metrics.CadenceLatency)
+	scope := common.NewPerTaskListScope(pp1.Domain, pp1.TaskList.GetName(), pp1.TaskList.GetKind(), h.metricsClient, metrics.FrontendPollForDecisionTaskScope).Tagged(metrics.GetContextTags(ctx)...)
+	scope.IncCounter(metrics.CadenceRequestsPerTaskList)
+	sw := scope.StartTimer(metrics.CadenceLatencyPerTaskList)
```
Comment (Groxx) [inline service/frontend/wrappers/metered/api_generated.go:324]: this might deserve a rollup value, as we'll lose domain-wide aggregates with this change

**Review by Groxx — CHANGES_REQUESTED** [review]
just blocking temporarily to make sure a rollup value is considered.
if we don't want/need the per-domain rollup, then LGTM, happy to unblock 👍

File: service/frontend/wrappers/metered/api_generated.go (hunk 1 above)
Comment (taylanisikdemir) [reply service/frontend/wrappers/metered/api_generated.go:324]: Agree. I also though that might be a problem and based on [this line](https://github.com/uber/cadence/blob/a711217807fb05102232b1dd08aa0f492cd6d114/common/metrics/defs.go#L2804) it looks like there's rollup defined for this metric.

Comment (Shaddoll) [reply service/frontend/wrappers/metered/api_generated.go:326]: UPDATED! Also fixed a bug in metrics package.

Comment (Groxx) [reply service/frontend/wrappers/metered/api_generated.go:326]: ah, just doing a manual non-TL-containing timer.  yea, that seems fine 👍

File: common/metrics/scope.go [hunk 2]
```diff
@@ -120,7 +120,7 @@ func (m *metricsScope) RecordHistogramValue(id int, value float64) {
 }
 
 func (m *metricsScope) Tagged(tags ...Tag) Scope {
-	domainTagged := false
+	domainTagged := m.isDomainTagged
 	tagMap := make(map[string]string, len(tags))
```
Comment (Groxx) [inline common/metrics/scope.go:124]: aaah.  yeah, this makes sense - otherwise we'd lose the value when we make a cached scope with a domain tag.  👍

**Review by Groxx — APPROVED** [review]
Yep, looks good now.  thank you!

What does the isDomainTagged value do, anyway?  The fix looks definitely correct, I'm just not familiar with what it actually does.

**Pull request merged**